.github/scripts/upload-to-gcs.sh

```
@@ -15,18 +15,22 @@ if [[ "${DRY_RUN:-false}" == "true" ]]; then
   exit 0
 fi
 
-echo "${GCS_SERVICE_ACCOUNT_KEY}" > /tmp/gcs-key.json
+GCS_KEY_FILE=$(mktemp)
+echo "${GCS_SERVICE_ACCOUNT_KEY}" > "${GCS_KEY_FILE}"
+chmod 600 "${GCS_KEY_FILE}"
+trap "rm -f '${GCS_KEY_FILE}'" EXIT
```

Shellcheck (0.11.0)

[warning] Use single quotes, otherwise this expands now rather than when signalled.

(SC2064)